doc/source/whatsnew/v1.2.0.rst

```diff
@@ -8,6 +8,16 @@ including other versions of pandas.
 
 {{ header }}
 
+.. warning::
+
+   The default argument ``engine=None`` to ``pd.read_excel`` is
```

Comment:
great if we think it's overkill than can remove before the release 

but can't hurt having it here